### PR TITLE
Reorganize horizon databases and tables

### PR DESCRIPTION
**Pull request by Tryneus**
This branch introduces a race condition when `auto_create_collection` is enabled by multiple horizon servers for the same project, but once ReQL supports it (rethinkdb/rethinkdb#5987), we can disambiguate name-conflicted tables.

There may still be some problems with the `schema` and `serve` commands, mostly because a lot of branches touched them recently, but I plan on testing them thoroughly in the near future.

---

This change is [Reviewable](https://reviewable.io/reviews/rethinkdb/horizon/685)

### CODE REVIEW

File: server/src/metadata/collection.js [hunk 1]
```diff
@@ -38,10 +29,23 @@ class Collection {
...
+      this.table = table.table;
+      this._table = table;
+      this._waiters.forEach((done) => this._table.on_ready(done));
+      this._waiters = [ ];
+    } else {
+      const old_table = this._table;
+      this.table = null;
+      this._table = null;
+
+      if (old_table) {
```
Comment (mlucy) [inline server/src/metadata/collection.js]: It seems like if we need to do cleanup when removing a table here, we need to do it in the branch above as well.

File: server/src/metadata/metadata.js [hunk 2]
```diff
@@ -353,14 +355,14 @@ class Metadata {
     error.check(this._collections.get(name) === undefined,
                 `Collection "${name}" already exists.`);
 
-    const collection = new Collection({ id: name, table: null }, this._db);
+    const collection = new Collection(name, null);
```
Comment (mlucy) [inline server/src/metadata/metadata.js:360]: Consider a comment.

File: server/src/schema/server_options.js [hunk 3]
```diff
@@ -3,7 +3,7 @@
 const Joi = require('joi');
 
 const server = Joi.object({
-  project_name: Joi.string().default('horizon'),
+  project_name: Joi.string().default('project'),
```
Comment (mlucy) [inline server/src/schema/server_options.js]: Consider changing this back to horizon.

**Comment by mlucy** [comment]
LGTM

**Comment by Tryneus** [comment]
Comments have been addressed, merging.

**Pull request merged**